exporter/trace/stackdriver/stackdriver.go

```diff
@@ -0,0 +1,290 @@
+		ctx := o.Context
+		if ctx == nil {
+			ctx = context.Background()
+		}
+		creds, err := google.FindDefaultCredentials(ctx, traceapi.DefaultAuthScopes()...)
+		if err != nil {
+			return nil, fmt.Errorf("Stackdriver: %v", err)
+		}
+		if creds.ProjectID == "" {
+			return nil, errors.New("Stackdriver: no project found with application default credentials")
+		}
+		o.ProjectID = creds.ProjectID
+	}
+	te, err := newTraceExporter(&o)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Exporter{
+		traceExporter: te,
+	}, nil
+}
+
+func newContextWithTimeout(ctx context.Context, timeout time.Duration) (context.Context, func()) {
+	if ctx == nil {
+		ctx = context.Background()
+	}
+	if timeout <= 0 {
+		timeout = defaultTimeout
+	}
+	return context.WithTimeout(ctx, timeout)
+}
+
+// RegisterBatchSpanProcessor registers e as BatchSpanProcessor.
+func (e *Exporter) RegisterBatchSpanProcessor() error {
+	opts := []trace.BatchSpanProcessorOption{}
+
+	if e.traceExporter.o.BatchDelayThreshold > 0 {
+		opts = append(opts, trace.WithScheduleDelayMillis(e.traceExporter.o.BatchDelayThreshold))
+	}
+	if e.traceExporter.o.MaxQueueSize > 0 {
+		opts = append(opts, trace.WithMaxQueueSize(e.traceExporter.o.MaxQueueSize))
+	}
+	if e.traceExporter.o.MaxExportBatchSize > 0 {
+		opts = append(opts, trace.WithMaxExportBatchSize(e.traceExporter.o.MaxExportBatchSize))
+	}
+
+	var err error
+	e.spanProcessor, err = trace.NewBatchSpanProcessor(e, opts...)
+	if err != nil {
+		return err
+	}
+	trace.RegisterSpanProcessor(e.spanProcessor)
+	return nil
+}
+
+// RegisterSimpleSpanProcessor registers e as SimpleSpanProcessor.
+func (e *Exporter) RegisterSimpleSpanProcessor() {
+	spanProcessor := trace.NewSimpleSpanProcessor(e)
+	trace.RegisterSpanProcessor(spanProcessor)
+}
+
+// ExportSpan exports a SpanData to Stackdriver Trace.
+func (e *Exporter) ExportSpan(ctx context.Context, sd *export.SpanData) {
+	if len(e.traceExporter.o.DefaultTraceAttributes) > 0 {
+		sd = e.sdWithDefaultTraceAttributes(sd)
+	}
+	e.traceExporter.ExportSpan(ctx, sd)
+}
+
+// ExportSpans exports a slice of SpanData to Stackdriver Trace in batch
+func (e *Exporter) ExportSpans(ctx context.Context, sds []*export.SpanData) {
+	e.traceExporter.ExportSpans(ctx, sds)
+}
+
+func (e *Exporter) sdWithDefaultTraceAttributes(sd *export.SpanData) *export.SpanData {
+	newSD := *sd
+	for k, v := range e.traceExporter.o.DefaultTraceAttributes {
+		switch val := v.(type) {
+		case bool:
+			newSD.Attributes = append(newSD.Attributes, key.New(k).Bool(val))
+		case int64:
+			newSD.Attributes = append(newSD.Attributes, key.New(k).Int64(val))
+		case float64:
+			newSD.Attributes = append(newSD.Attributes, key.New(k).Float64(val))
+		case string:
+			newSD.Attributes = append(newSD.Attributes, key.New(k).String(val))
+		}
+	}
+	newSD.Attributes = append(newSD.Attributes, sd.Attributes...)
+	return &newSD
+}
+
+// Shutdown unregisters spanProcessor.
+func (e *Exporter) Shutdown() {
+	e.spanProcessor = nil
+}
```

Comment:
Did you forget to change this to unregister?

Reply:
I just realized the function. Added the process before the line.